weave

```diff
@@ -696,16 +703,16 @@ init_bridge() {
 
     ip link add name $BRIDGE type bridge
 
-    # Derive the bridge MAC from the system (aka bios) UUID, or,
-    # failing that, the hypervisor UUID. Elsewhere we in turn derive
-    # the peer name from that, which we want to be stable across
-    # reboots but otherwise unique. The system/hypervisor UUID fits
-    # that bill, unlike, say, /etc/machine-id, which is often
-    # identical on VMs created from cloned filesystems. If we cannot
-    # determine the system/hypervisor UUID we just generate a random MAC.
-    MAC=$(mac_from_file /sys/class/dmi/id/product_uuid) || \
-        MAC=$(mac_from_file /sys/hypervisor/uuid) || \
-        MAC=$(random_mac)
+    # Derive the bridge MAC from concatenated machine-id (either systemd or
+    # dbus), the system (aka bios) UUID and the hypervisor UUID. If we cannot
+    # determine all of them we just generate a random MAC.
+    # Elsewhere we in turn derive the peer name from the bridge MAC, which we
+    # want to be stable across reboots but otherwise unique.
+
+    local uid="$MACHINE_ID"
```

Comment:
I know. I was wondering whether there is any particular reason for not using `local` in the weave script besides brevity?

Also, how do we denote that the scope of a variable is local? By lower case? If yes, then there are some inconsistencies, e.g. https://github.com/weaveworks/weave/blob/master/weave#L1414 vs https://github.com/weaveworks/weave/blob/master/weave#L473.